projects/components/src/header/page/page-header.component.ts

```diff
@@ -23,26 +26,67 @@ import { NavigableTab } from '../../tabs/navigable/navigable-tab';
       class="page-header"
       [class.bottom-border]="!this.tabs?.length"
     >
-      <div [ngClass]="this.contentAlignment">
-        <div class="breadcrumb-container">
-          <ht-breadcrumbs [breadcrumbs]="breadcrumbs"></ht-breadcrumbs>
-          <div class="title" *ngIf="this.titlecrumb$ | async as titlecrumb">
-            <ht-icon
-              class="icon"
-              *ngIf="titlecrumb.icon && breadcrumbs.length <= 1"
-              [icon]="titlecrumb.icon"
-              [label]="titlecrumb.label"
-              size="${IconSize.Large}"
-            ></ht-icon>
-
-            <ht-label [label]="titlecrumb.label"></ht-label>
-            <ht-beta-tag *ngIf="this.isBeta" class="beta"></ht-beta-tag>
+      <!-- If PageTimeRange feature flag is enabled, consumer of this component must specify the type of content being
+           projected, primary or secondary -->
+      <div
+        *htIfFeature="'${ApplicationFeature.PageTimeRange}' | htFeature; else noTimeRangeHeaderLayoutTemplate"
+        [ngClass]="this.contentAlignment"
+      >
+        <div class="primary-content">
+          <div class="breadcrumb-container">
+            <ht-breadcrumbs [breadcrumbs]="breadcrumbs"></ht-breadcrumbs>
+            <div class="title" *ngIf="this.titlecrumb$ | async as titlecrumb">
+              <ht-icon
+                class="icon"
+                *ngIf="titlecrumb.icon && breadcrumbs.length <= 1"
+                [icon]="titlecrumb.icon"
+                [label]="titlecrumb.label"
+                size="${IconSize.Large}"
+              ></ht-icon>
+
+              <ht-label [label]="titlecrumb.label"></ht-label>
+              <ht-beta-tag *ngIf="this.isBeta" class="beta"></ht-beta-tag>
+            </div>
           </div>
+          <ng-container *ngTemplateOutlet="this.primaryRowContent?.templateRef"></ng-container>
+          <ht-user-specified-time-range-selector
+            class="time-range"
+            *htIfFeature="'${ApplicationFeature.NavigationRedesign}' | htFeature; else globalTimeRange"
+          ></ht-user-specified-time-range-selector>
+
+          <ng-template #globalTimeRange>
```

Comment:
please use `Template` suffix for template names. applicable at other places as well.

Reply:
done